### PR TITLE
serve javascript and CSS from nginx

### PR DESCRIPTION
**Pull request by chris48s**
* Run `collectstatic` during the docker build
* Run nginx on the container
* Serve static files with nginx
* Leave the `/media` (uploads) dir on S3 and continue to serve it through CloudFront

### Logs

I've told supervisord, gunicorn and nginx to log to stdout only so we're not retaining any logs. This is not what I'd usually want to do, but it does replicate the existing setup:
https://github.com/okfn/website/blob/a1317eeacd737663a660ed581a99c624a125190a/Dockerfile#L29-L33

### Cache Headers

I'm serving the static files with a `no-cache` (note: not `no-store`) header
https://csswizardry.com/2019/03/cache-control-for-civilians/#no-cache
This allows the browser to serve the file from cache, but it has to check whether the content has changed first. Nginx serves an `Etag` header as well as a `Last-Modified`, so this enables most sensible browsers to cache in a fairly smart way (based on the etag) while ensuring we aren't stuck with stale versions in the wild we can't control. While we are just writing static files to the same location (`style.css`, `scripts.js`) every time we `collectstatic` this should be a safe approach. It might be useful to revisit this and switch to fingerprinted assets with long `max-age` headers using django-pipeline or something but I think its better to review that after we've done some work on updating the application code and reviewed the kitchen-sink of css/js we are writing to `staticfiles`

~TODO: Build/push a DockerHub image, deploy to staging and test~

### CODE REVIEW

File: deployment/gunicorn.config.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+bind = '127.0.0.1:5000'
+
+workers = 2
```
Comment (chris48s) [inline deployment/gunicorn.config.py:3]: I think we were actually running with only one worker before but we should be able to run 2 or 3 without too much problem even with only a single core.

From https://docs.gunicorn.org/en/stable/run.html#commonly-used-arguments

> This number should generally be between 2-4 workers per core in the server.

File: deployment/nginx.conf [hunk 2]
```diff
@@ -0,0 +1,24 @@
+server {
+    listen 80;
+
```
Comment (chris48s) [inline deployment/nginx.conf:3]: I'm slightly unclear if we also need a `server_name` statement here or not. I think I'm just going to try it on staging and see what happens.

**Comment by amercader** [comment]
This all looks sensible.
Regarding the logs I think as long as we can check them via `kubectl` is fine for now. If we need to debug something historically we can start storing them somewhere.
For reference here's how to access them:

```
kubectl -n production logs -l app=okfn -f
```

File: deployment/nginx.conf (hunk 2 above)
Comment (chris48s) [reply deployment/nginx.conf:3]: seems to be fine without it :shrug:

**Comment by chris48s** [comment]
Aah I hadn't realised that kubectl would catch the logs we send to stdout/stderr :+1:  TIL :mortar_board: 

This is now deployed on https://next.okfn.org/ (and was quicker/less error prone to deploy :tada: ). Seems to all be fine, but feel free to have a poke about and see if you can find any issues.

I'll have a look at deploying to production w/c 17th.

**Comment by chris48s** [comment]
TODO: once this is the standard build for prod/staging, clear up stuff we don't need any more from S3

**Pull request merged**